src/main/scala/com/microsoft/hyperspace/actions/RefreshDeleteAction.scala

```diff
@@ -57,15 +57,16 @@ class RefreshDeleteAction(
    */
   final override def validate(): Unit = {
     super.validate()
+
+    if (deletedFiles.isEmpty) {
+      throw NoChangesDetected("Refresh aborted as no deleted source data file found.")
+    }
+
```

Comment:
for reviewers: this order is changed because if there are no deleted files, we don't really need to go further and fail the operation for lineage column missing.